mlrun/datastore/sources.py

```diff
@@ -393,6 +377,7 @@ def to_step(
         attributes.pop("additional_filters", None)
         if context:
             attributes["context"] = context
+        additional_filters = transform_list_filters_to_tuple(additional_filters)
```

Comment:
10 lines down, we have `additional_filters=self.additional_filters or additional_filters`. What if we end up getting the additional filters from self and not from the parameter?

Reply:
in self additional filters saved as tuple, you can see the transformation I did in `from_dict` function and `__init__`